requirements/customer/scheduling-algorithm.tex

```diff
@@ -0,0 +1,48 @@
+\subsection{Scheduling Algorithm}
+This section describes how the central server distributes jobs to work machines: the \textit{scheduling algorithm}.
+Put briefly, the scheduling algorithm takes a list of jobs and decides which ones to run on the available work machines.
+The scheduling algorithm has the following goals:
+\begin{itemize}
+\item Keep work machines busy: there should be no idle work machines if there are pending jobs that could be run on them.
+\item Consider job priorities: jobs with high priorities should be worked on first.
+\item Consider time of job submission: jobs that were submitted first should be worked on first (if priorities are equal).
+\end{itemize}
+The scheduling algorithm operates under the following conditions:
+\begin{itemize}
+\item Jobs arrive at unforeseen times.
+\item Time required to complete jobs is unknown.
+\item Jobs cannot be transferred from one work machine to another.
+\item Jobs cannot be canceled and restarted.
+\item Job preemption is limited: if a job is paused it has to be kept in memory.
```

Comment:
I do not agree with this point, but I am not sure if that needs to be addressed here or at a later stage. We do not consider swap to be part of the memory at all, it is used entirely for paused jobs.

Reply:
Oh you have explained it below ... Maybe it is better to just say `kept in memory or in swap`